src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+const isValid = (...edges:number[]): boolean => {
+  return edges.some((side) => side <= 0);
+};
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (isValid(sideA, sideB, sideC)) {
+      throw new Error('SIDE CAN\'T BE LESS THEN 0');
+    }
+
+    if (sideA + sideB <= sideC
+      || sideA + sideC <= sideB
+      || sideB + sideC <= sideA) {
+      throw new Error('SUM OF 2 SIDES CAN\'T BE GREATER THEN THIRD SIDE');
+    }
+  }
 
+  getArea():number {
+    const halfPerimeter = (this.sideA + this.sideB + this.sideC) / 2;
+    const area = Math.sqrt(halfPerimeter * (halfPerimeter - this.sideA)
+      * (halfPerimeter - this.sideB) * (halfPerimeter - this.sideC));
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you round down to hundredths on each getArea, maybe better create a separate method for it?